From 0b9d383131e60051cc45d91f13963cd2443547f7 Mon Sep 17 00:00:00 2001
From: ganeshi4u <ganeshvarmai4u@gmail.com>
Date: Mon, 2 Dec 2019 15:46:19 +0100
Subject: [PATCH] deviceinfo: Add device name inside settings firmwareversion
 layout

Squashed with the following:
Author: Adithya R <gh0strider.2k18.reborn@gmail.com>
Date:   Mon Nov 9 11:15:02 2020 +0530

    Settings: Read device codename from ro.product.device

     * On unified builds for devices with different codenames, init extension sets the
       appropriate codename. Display that codename instead of the one set at build time.

    Change-Id: I093414a79153323cff9140879ec582354861a1e7

Change-Id: I61fd4d77375df019da94ed7fca71da157dd0f415
---
 res/values/calyx_strings.xml                  |  1 +
 res/xml/firmware_version.xml                  |  8 ++++
 .../AboutDeviceNamePreferenceController.java  | 45 +++++++++++++++++++
 3 files changed, 54 insertions(+)
 create mode 100644 src/com/android/settings/deviceinfo/firmwareversion/AboutDeviceNamePreferenceController.java

diff --git a/res/values/calyx_strings.xml b/res/values/calyx_strings.xml
index bb159a7b3b..375d90097c 100644
--- a/res/values/calyx_strings.xml
+++ b/res/values/calyx_strings.xml
@@ -16,6 +16,7 @@
 -->
 <resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
     <string name="calyxos_version">CalyxOS version</string>
+    <string name="about_device_name">Device</string>
 
     <string name="private_dns_mode_cloudflare" translatable="false">Cloudflare DNS</string>
 
diff --git a/res/xml/firmware_version.xml b/res/xml/firmware_version.xml
index b5a61cbcdd..97bbd528ad 100644
--- a/res/xml/firmware_version.xml
+++ b/res/xml/firmware_version.xml
@@ -38,6 +38,14 @@
         settings:enableCopying="true"
         settings:controller="com.android.settings.deviceinfo.firmwareversion.CalyxOSVersionPreferenceController"/>
 
+    <!-- Device name -->
+    <Preference
+        android:key="about_device_name"
+        android:title="@string/about_device_name"
+        android:summary="@string/summary_placeholder"
+        settings:enableCopying="true"
+        settings:controller="com.android.settings.deviceinfo.firmwareversion.AboutDeviceNamePreferenceController"/>
+
     <!-- Security patch -->
     <Preference
         android:key="security_key"
diff --git a/src/com/android/settings/deviceinfo/firmwareversion/AboutDeviceNamePreferenceController.java b/src/com/android/settings/deviceinfo/firmwareversion/AboutDeviceNamePreferenceController.java
new file mode 100644
index 0000000000..9be2c1f38b
--- /dev/null
+++ b/src/com/android/settings/deviceinfo/firmwareversion/AboutDeviceNamePreferenceController.java
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2019 ArrowOS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.settings.deviceinfo.firmwareversion;
+
+import android.content.Context;
+import android.os.SystemProperties;
+
+import com.android.settings.R;
+import com.android.settings.core.BasePreferenceController;
+
+public class AboutDeviceNamePreferenceController extends BasePreferenceController {
+
+    private static final String TAG = "AboutDeviceNameCtrl";
+
+    private static final String KEY_DEVICE_NAME_PROP = "ro.product.device";
+
+    public AboutDeviceNamePreferenceController(Context context, String key) {
+        super(context, key);
+    }
+
+    @Override
+    public int getAvailabilityStatus() {
+        return AVAILABLE;
+    }
+
+    @Override
+    public CharSequence getSummary() {
+        return SystemProperties.get(KEY_DEVICE_NAME_PROP,
+                mContext.getString(R.string.unknown));
+    }
+}
-- 
2.37.3

